lib/core/base/virtual-node.js

```diff
@@ -76,6 +76,15 @@ class VirtualNode {
 		return this.actualNode.hasAttribute(attrName);
 	}
 
+	/**
+	 * Get the input type. This is more complicated when you don't have an
+	 * actualNode to reference.
+	 * @return {String|undefined}
+	 */
+	get elementType() {
```

Comment:
This might get confused with `nodeType`. Can we have a different name for this?

Reply:
Sure. I'd like it to be obvious that we're accessing JavaScript property values, and be consistent with other properties we might need (like `attributes`). I think we should also make individual property lookup functions and not a generic function like jQuery's `props` so anyone overriding the class definition knows exactly which props are needed. 

Some thoughts:

- `propType` (might get confusing with React)
- `typeProp`
- `getTypeProp` or `getPropType` 
-  just `type`

Reply:
How about `inputType`?

Reply:
My only concern with that is that `input`s are not the only element with a `type` property. `object` elements also have one https://developer.mozilla.org/en-US/docs/Web/HTML/Element/object.

Reply:
`.type` makes sense to me, since that's the native property too.

Reply:
How should we address the fact that `nodeType`, `nodeName` and `id` are prefixed with `element` but `type` is not?

Reply:
Yikes. That's no good.

To avoid adding an inconsistent property name, I think `elementType` might be the best option. We probably want to consider changing this naming convention in v4 tho.

Reply:
What's wrong with the naming convention? We can change anything right now before 3.3 is released.

Reply:
We decided to namespace all JavaScript node properties under the `.props` object named the same as their Node counterpart.